### PR TITLE
[Quantic] RGA Show More/Show Less buttons

### PR DESCRIPTION
**Pull request by erocheleau**
[SFINT-5428]

Until the scratch org expires, you can try it out here:

https://enterprise-enterprise-2467-dev-ed.scratch.my.site.com/examples/s/relevant-generative-answering-example#q=how%20to%20resolve%20netflix%20connection%20with%20tivo&tab=All
(Valid until the 25th)

Adding a new option to the QuanticGeneratedAnswer component: `collapsible`.
This option is `false` by default. 

When turned on, this option will cap the size of the generated answer text to a maximum of 250px.
A new control will be added to expand and/or collapse the answer that was generated.

Some details on the behavior:

- While the answer is collapsed, the rephrase, feedback and citations controls are not visible.
- When expanding the answer, the above controls appear.
- The state of expanded/collapsed will reset on each query, rephrase of RGA as designed.
- If the answer is actually shorter than 250px nothing will change to the current behavior of RGA.
  - The rephrase, feedback, and citations components will all be visible.
  - No show more/less button should be visible.
- The new button should render left of the RGA disclaimer (so on a row below the Rephrase and citations)
- The new button should work correctly when the footer is `multiline`
- The cap to the answer height should look "clean" with a fade out, etc.

See videos and screenshots below for examples.

1. The component and the new control should be responsive:

https://github.com/coveo/ui-kit/assets/2488155/653fc524-3916-4961-8fa5-61f9b1899f94

2. General function of the component:

https://github.com/coveo/ui-kit/assets/2488155/75cbb774-9da4-43c5-befd-3b41ef8caccb

[SFINT-5428]: https://coveord.atlassian.net/browse/SFINT-5428?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Also found a fix for : SFINT-5469

### CODE REVIEW

**Comment by mmitiche** [comment]
I will start with a high level UX concern: 

## Issue:

When the component is collapsed and the answer is streaming, it may seem confusing to understand what's happening with the component  as the generated answer will be collapsed and the end user will not even know that the answer is still streaming  and no controls are displayed to interact with the component.  

This confusing situation occurred for maybe one second or less in the demo showed in the PR, but if the generated answer is super long and it takes more  time to stream this confusing moment will take longer:

https://github.com/coveo/ui-kit/assets/86681870/710fe96f-7754-4b50-8218-5accb2422b07


I found this particular moment a bit confusing: 

Screenshot 2024-04-18 at 1 51 48 PM
As an end user when I see this, I can't even know if the answer completed streaming or not, I can't interact with the component, I'm kinda clueless what's happening.


## Suggestion

I suggest having a mechanism that would inform the end user that the generated answer is still streaming even if the generated answer is collapsed to avoid confusion.
I'm no UX, but just to explain my idea, something like that would explain better what's going on in the RGA component:
Screenshot 2024-04-18 at 1 46 57 PM
Notice the label `streaming...` at the bottom of the component, this label could be an icon or anything that would just inform the user that answer is being generated.
Once the answer generated, the label would disappear and the component controls will appear.


What do you think @adroletCoveo  , @erocheleau , @jelmedini ?

**Comment by mmitiche** [comment]
> > I will start with a high level UX concern:
> > ## Issue:
> > When the component is collapsed and the answer is streaming, it may seem confusing to understand what's happening with the component as the generated answer will be collapsed and the end user will not even know that the answer is still streaming and no controls are displayed to interact with the component.
> > This confusing situation occurred for maybe one second or less in the demo showed in the PR, but if the generated answer is super long and it takes more time to stream this confusing moment will take longer:
> > Screen.Recording.2024-04-18.at.1.37.38.PM.mov
> > I found this particular moment a bit confusing:
> > Screenshot 2024-04-18 at 1 51 48 PM As an end user when I see this, I can't even know if the answer completed streaming or not, I can't interact with the component, I'm kinda clueless what's happening.
> > ## Suggestion
> > I suggest having a mechanism that would inform the end user that the generated answer is still streaming even if the generated answer is collapsed to avoid confusion. I'm no UX, but just to explain my idea, something like that would explain better what's going on in the RGA component: Screenshot 2024-04-18 at 1 46 57 PM Notice the label `streaming...` at the bottom of the component, this label could be an icon or anything that would just inform the user that answer is being generated. Once the answer generated, the label would disappear and the component controls will appear.
> > What do you think @adroletCoveo , @erocheleau , @jelmedini ?
> 
> @MehdiMitiche I agree this is the issue that we raised also, I was thinking of the same solution, just waiting to validate this with Audrey, she is off this week

Perfect 👌

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js [hunk 1]
```diff
@@ -122,6 +133,14 @@ export default class QuanticGeneratedAnswer extends LightningElement {
   ariaLiveMessage;
   /** @type {boolean} */
   hasInitializationError = false;
+  /** @type {boolean} */
+  _answerDisplayUpdated = false;
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:137]: Unless I'm mistaken, I don't see that this is used anywhere?

File: packages/quantic/force-app/examples/main/lwc/exampleRelevantGenerativeAnswering/exampleRelevantGenerativeAnswering.html [hunk 2]
```diff
@@ -130,6 +130,7 @@
                   engine-id={engineId}
                   answer-style={answerStyle}
                   multiline-footer={multilineFooter}
+                  collapsible="true"
```
Comment (mmitiche) [inline packages/quantic/force-app/examples/main/lwc/exampleRelevantGenerativeAnswering/exampleRelevantGenerativeAnswering.html:133]:
```suggestion
                  collapsible
```

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js [hunk 3]
```diff
@@ -85,6 +87,13 @@ export default class QuanticGeneratedAnswer extends LightningElement {
    * @default {false}
    */
   @api multilineFooter;
+  /**
+   * Whether the generated answer should be collapsible when it exceeds the maximum height of 250px.
+   * @api
+   * @type {boolean}
+   * @default {false}
+   */
+  @api collapsible = false;
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:96]: Just a minor suggestion for the name, `isCollapsible` sounds slightly better for me as a boolean proeprty..

File: packages/quantic/cypress/jsconfig.json [hunk 4]
```diff
@@ -1,3 +1,4 @@
 {
-  "include": ["../../node_modules/cypress", "./**/*.js"]
+  "include": ["../../node_modules/cypress", "./**/*.js"],
+  "typeAcquisition": {"include": ["cypress"]}
```
Comment (mmitiche) [inline packages/quantic/cypress/jsconfig.json:3]: So this fixing the Cypress typing errors?

very nice 👌

**Review by mmitiche — COMMENTED** [review]
I didn't yet review the tests, but the logic of the solution is neat not complicated and efficient, nice job 👍

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/templates/generatedAnswer.css [hunk 5]
```diff
@@ -30,6 +28,48 @@
   vertical-align: baseline;
 }
 
+.generated-answer__answer--collapsed {
+  max-height: var(--maxHeight, 250px);
+}
+
+.generated-answer__answer--expanded {
+  max-height: var(100%);
```
[the remark this answers is not in the document]
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/templates/generatedAnswer.css:36]: haha yes, there used to be something else and I removed it 😂

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/templates/generatedAnswer.css [hunk 6]
```diff
@@ -30,6 +28,48 @@
...
+
+.generated-answer__answer--collapsed::after {
+  clip-path: polygon(0 0, 100% 0, 100% 100%, 0% 100%);
+  opacity: 1;
+  visibility: visible;
+}
+
+.generated-answer__answer--expanded::after {
+  clip-path: polygon(0 100%, 100% 100%, 100% 100%, 0% 100%);
+  opacity: 0.2;
```
[the remark this answers is not in the document]
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/templates/generatedAnswer.css:65]: 🤷 I was trying to start from the exact same thing as the Smart snippet one, so I copied it from what you had done with the Smart Snippet because it looked nice! lol

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js [hunk 7]
```diff
@@ -141,6 +160,13 @@ export default class QuanticGeneratedAnswer extends LightningElement {
 
   renderedCallback() {
     initializeWithHeadless(this, this.engineId, this.initialize);
+    if (this.collapsible) {
+      const answerElementHeight = this.isStreaming
+        ? this.generatedAnswerElementHeight + 50
```
[the remark this answers is not in the document]
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:165]: I thought I added a comment in the code to explain that, I'll re-add it.

In short it's because we're still streaming so we "know" the answer isn't yet complete. To avoid some jankyness, I pre-suppose the answer will at least be +50 px in height in order to start displaying the fade-out effect sooner.

This +50 disappears once the answer is no longer streaming so we can take the real height at this point.

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js (hunk 1 above)
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:137]: You're correct this was a remnant of the previous way I was handling the collapse.

I've had a lot of trouble while writing the tests because the answer is never "streaming" when we write our e2e tests because we just mock the "final" message.

So anything that was based off of `isStreaming` wasn't working correctly but just because of our tests.

File: packages/quantic/force-app/examples/main/lwc/exampleRelevantGenerativeAnswering/exampleRelevantGenerativeAnswering.html (hunk 2 above)
Comment (erocheleau) [reply packages/quantic/force-app/examples/main/lwc/exampleRelevantGenerativeAnswering/exampleRelevantGenerativeAnswering.html:133]: Yea actually I think I'll remove this since the default should be false?

The property can still be found and used in the test version of the component, but the default search page shouldn't have a collapsible RGA. 

Thanks I forgot to remove it.

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js (hunk 3 above)
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:96]: Even when it's going to be used in the HTML?

`<c-quantic-generated-answer is-collapsible` ?

File: packages/quantic/cypress/jsconfig.json (hunk 4 above)
Comment (erocheleau) [reply packages/quantic/cypress/jsconfig.json:3]: Yes! I finally found a solution! This and the change to the tsconfig

I think this one is the major fix `"include": ["../node_modules/cypress"` in the tsconfig that's under the cypress folder.

I believe this means the cypress code is included in the steps of compiling the TS for the cypress tests, so it replaces the default jest types that are brought from the other files? 🤷 

It comes from the Cypress doc and their example repo 
https://docs.cypress.io/guides/tooling/typescript-support#Clashing-types-with-Jest
https://github.com/cypress-io/cypress-and-jest-typescript-example

**Comment by SimonMilord** [comment]
Maybe this is nitpicking, but I wonder why is the fade in effect on the last line when you expand? Wouldn't it look better if it was applied to all the text being revealed? or even not at all? Not sure if Im the only one.

see:

https://github.com/coveo/ui-kit/assets/73316533/8b7504c7-66cb-45e3-9671-19a297f10107

File: packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts [hunk 8]
```diff
@@ -63,6 +64,17 @@ const genQaMessageTypePayload = {
   finishReason: 'COMPLETED',
 };
 
+const testLongText = `Call me Ishmael. Some years ago—never mind how long precisely—having little or no money in my purse, and nothing particular to interest me on shore, I thought I would sail about a little and see the watery part of the world. It is a way I have of driving off the spleen and regulating the circulation. Whenever I find myself growing grim about the mouth; whenever it is a damp, drizzly November in my soul; whenever I find myself involuntarily pausing before coffin warehouses, and bringing up the rear of every funeral I meet; and especially whenever my hypos get such an upper hand of me, that it requires a strong moral principle to prevent me from deliberately stepping into the street, and methodically knocking people’s hats off—then, I account it high time to get to sea as soon as I can. This is my substitute for pistol and ball. With a philosophical flourish Cato throws himself upon his sword; I quietly take to the ship. There is nothing surprising in this. If they but knew it, almost all men in their degree, some time or other, cherish very nearly the same feelings towards the ocean with me.
```
Comment (SimonMilord) [inline packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts:67]: more of a lord of the ring guy, but Ill allow it 🤣

File: packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer-selectors.ts [hunk 9]
```diff
@@ -101,8 +102,8 @@ export const GeneratedAnswerSelectors: GeneratedAnswerSelector = {
     GeneratedAnswerSelectors.get().find(
       `c-quantic-radio-buttons-group [data-cy="${label}"]`
     ),
-  generatedAnswerFooter: () =>
-    GeneratedAnswerSelectors.get().find('[data-cy="generated-answer__footer"]'),
+  generatedAnswerFooterRow: () =>
+    GeneratedAnswerSelectors.get().find('.generated-answer__footer-row'),
```
Comment (SimonMilord) [inline packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer-selectors.ts:106]: why not? best practice and more precise according to the cypress doc, plus the other selectors already use it

```suggestion
    GeneratedAnswerSelectors.get().find('[data-cy="generated-answer__footer-row"]'),
```

File: packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts [hunk 10]
```diff
@@ -238,7 +251,72 @@ describe('quantic-generated-answer', () => {
...
+              Expect.displayGeneratedAnswerToggleCollapseButton(true);
+              Expect.generatedAnswerToggleCollapseButtonContains('Show more');
+              Expect.displayCitations(false);
+              Expect.displayRephraseButtons(false);
+              Expect.displayDisclaimer(true);
+            });
+
+            scope('when clicking the toggle collapse button', () => {
+              Actions.clickToggleCollapseButton();
+              // TODO: Expect analytics to be sent
```
Comment (SimonMilord) [inline packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts:283]: these I assume will be added in a seperate PR?

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js [hunk 11]
```diff
@@ -319,6 +350,39 @@ export default class QuanticGeneratedAnswer extends LightningElement {
...
+    return this.template.querySelector('.generated-answer__answer');
+  }
+
+  /**
+   * Returns the generated answer height.
+   * @returns {number}
+   */
+  get generatedAnswerElementHeight() {
+    // @ts-ignore
+    return getAbsoluteHeight(this.generatedAnswerElement?.firstChild);
```
Comment (SimonMilord) [inline packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:367]: we need this ts-ignore?

File: packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer-selectors.ts (hunk 9 above)
Comment (erocheleau) [reply packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer-selectors.ts:106]: Sure but I don't agree it's a best practice though. If the css class changes, I want to be notified by a potential test selector failing. If you use data-cy everywhere, you can randomly change the CSS classes of your elements and the tests still pass...

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js (hunk 11 above)
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:373]: There's a bunch of conflicting types here, `firstChile` returns a `ChildNode` instead of an HTMLElement but it doesn't matter etc.

**Comment by erocheleau** [comment]
> Maybe this is nitpicking, but I wonder why is the fade-in effect on the last line when you expand? Wouldn't it look better if it was applied to all the text being revealed? or even not at all? personally I think it looks kinda weird, not sure if Im the only one.

I'll see if we want to change it later. It's only because there's a CSS transition on the element height I think.

**Comment by erocheleau** [comment]
Updated the video demonstrating the feature

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js [hunk 12]
```diff
@@ -349,9 +417,15 @@ export default class QuanticGeneratedAnswer extends LightningElement {
...
+    let collapsedClass = '';
+    if (this._exceedsMaximumHeight) {
+      collapsedClass = this.isExpanded
+        ? 'generated-answer__answer--expanded'
+        : 'generated-answer__answer--collapsed';
+    }
     return `generated-answer__answer ${
       this.isStreaming ? 'generated-answer__answer--streaming' : ''
-    }`;
+    } ${collapsedClass}`;
```
Comment (SimonMilord) [inline packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:428]:
```suggestion
    let collapsibleClass = '';
    if (this._exceedsMaximumHeight) {
      collapsibleClass = this.isExpanded
        ? 'generated-answer__answer--expanded'
        : 'generated-answer__answer--collapsed';
    }
    return `generated-answer__answer ${
      this.isStreaming ? 'generated-answer__answer--streaming' : ''
    } ${collapsibleClass}`;
```

maybe I would name this collapsibleClass since it covers both cases where it can be collapsed or expanded? (A bit less confusing)

**Review by SimonMilord — APPROVED** [review]
Seems GTM! Logic looks sound and I think all the test cases seems to be covered. 
Nice job! 🎉

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js (hunk 12 above)
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js]: renamed it to `collapsedStateClass`

File: packages/quantic/cypress/tsconfig.json [hunk 13]
```diff
@@ -10,5 +10,5 @@
     "resolveJsonModule": true,
     "esModuleInterop": true
   },
-  "include": ["**/*.ts", "**/*.js"]
+  "include": ["../node_modules/cypress", "**/*.ts", "**/*.js"]
```
Comment (tedre191) [inline packages/quantic/cypress/tsconfig.json:13]: Not sure why this is required ? Feels weird to have to include something from node_modules here.
Just an initial reaction, this might be the right thing to do.

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js [hunk 14]
```diff
@@ -303,6 +331,15 @@ export default class QuanticGeneratedAnswer extends LightningElement {
     }
   };
 
+  handleToggleCollapseAnswer() {
+    if (this.state?.expanded) {
```
Comment (tedre191) [inline packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:335]: This looks like a nice opportunity for a ternary, up to you if you want to change it; I just like ternary (I might be the only one 😅 )

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js [hunk 15]
```diff
@@ -323,22 +387,26 @@ export default class QuanticGeneratedAnswer extends LightningElement {
     return this?.state?.citations;
   }
 
-  get hasCitations() {
-    return !!this.citations.length;
+  get shouldDisplayCitations() {
+    return !!this.citations.length && !this.isAnswerCollapsed;
```
Comment (tedre191) [inline packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:391]: Never been a fan of `!!this.citations.length`, or maybe create a variable `const citationIsEmpty = !![...]` to make the return statement simpler to understand ? It's not too bad but complex binary operations in a return statement like are sometime hard to follow.

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js [hunk 16]
```diff
@@ -323,22 +387,26 @@ export default class QuanticGeneratedAnswer extends LightningElement {
...
   }
 
   get isStreaming() {
     return this?.state?.isStreaming;
   }
 
   get shouldDisplayActions() {
```
Comment (tedre191) [inline packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:398]: I would add a comment here explaining the logic of the binary operation, otherwise it can be hard to grasp at first glance.

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js [hunk 17]
```diff
@@ -323,22 +387,26 @@ export default class QuanticGeneratedAnswer extends LightningElement {
...
-    return this.isVisible && !this.isStreaming;
+    return this.isVisible && !this.isStreaming && !this.isAnswerCollapsed;
   }
 
   get isVisible() {
     return this.state.isVisible;
   }
 
+  get isAnswerCollapsed() {
```
Comment (tedre191) [inline packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:406]: Same here

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js [hunk 18]
```diff
@@ -420,7 +496,41 @@ export default class QuanticGeneratedAnswer extends LightningElement {
...
   get disclaimerCssClass() {
-    return `slds-var-m-top_small slds-col slds-size_1-of-1 slds-text-color_weak slds-text-body_small ${this.multilineFooter ? '' : 'slds-text-align_right'}`;
+    return `generated-answer__disclaimer slds-grid_vertical-align-center slds-col slds-text-color_weak slds-text-body_small ${this.multilineFooter ? '' : 'slds-text-align_right'}`;
+  }
+
+  get toggleCollapseAnswerIcon() {
+    return this.isAnswerCollapsed ? 'utility:chevrondown' : 'utility:chevronup';
+  }
+
+  get shouldShowCollapseGeneratingMessage() {
```
Comment (tedre191) [inline packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:506]: Same here, comment to explain the different cases and the operation

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js [hunk 19]
```diff
@@ -420,7 +496,41 @@ export default class QuanticGeneratedAnswer extends LightningElement {
...
+    return (
+      this.collapsible &&
+      this.isVisible &&
+      this.isStreaming &&
+      this._exceedsMaximumHeight
+    );
+  }
+
+  get shouldShowToggleCollapseAnswer() {
+    return (
```
Comment (tedre191) [inline packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:516]: Same here, comment to explain the different cases and the operation

**Review by tedre191 — APPROVED** [review]

File: packages/quantic/cypress/tsconfig.json (hunk 13 above)
Comment (erocheleau) [reply packages/quantic/cypress/tsconfig.json:13]: Yea this is a "fix" to something weird, because we both use Jest and Cypress in the ui-kit repo, and they both export stuff globally like `describe, it, expect, etc.` it causes conflict in the intellisense of vscode.

This is a fix suggested by Cypress to resolved these conflicts. Notice that this `tsconfig` is under the /cypress folder, so the ts intellisense resolves it's settings by considering the closest tsconfig, so by including cypress here directly, it makes sure that the globally exported stuff by cypress are considered "over" the globally exported stuff by Jest.

Long explanation just to say this fixes a small technical issue with autocomplete and stuff in the cypress tests :)

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js (hunk 14 above)
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:335]: Modified the eslint file to allow ternary calls like that.

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js (hunk 16 above)
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:398]: Really, to me the comment 
`// Only display actions when the answer is visible, not while streaming, and not collapsed.` is longer to read than the true/false :D
